ghost/member-attribution/lib/attribution.js

```diff
@@ -80,21 +88,52 @@ class AttributionBuilder {
 
     /**
      */
-    constructor({urlTranslator}) {
+    constructor({urlTranslator, referrerTranslator}) {
         this.urlTranslator = urlTranslator;
+        this.referrerTranslator = referrerTranslator;
     }
 
     /**
      * Creates an Attribution object with the dependencies injected
      */
-    build({id, url, type}) {
+    build({id, url, type, referrerSource, referrerMedium, referrerUrl}) {
         return new Attribution({
             id,
             url,
-            type
+            type,
+            referrerSource,
+            referrerMedium,
+            referrerUrl
         }, {urlTranslator: this.urlTranslator});
     }
 
+    getReferrerAttributionData({history}) {
+        if (history.length === 0) {
+            return {
+                referrerSource: null,
+                referrerMedium: null,
+                referrerUrl: null
+            };
+        }
+
+        for (const item of history) {
+            const {referrerSource, referrerMedium, referrerUrl} = this.referrerTranslator.getAttributionData(item);
+            if (referrerSource && referrerMedium) {
+                return {
+                    referrerSource,
+                    referrerMedium,
+                    referrerUrl
+                };
+            }
```

Comment:
I'm wondering here that when both `item.referrerSource` and `item.referrerMedium` are null/undefined, that we still get default values. So if you visit
`https://ghost.site/my-post?rel=this-is-referred`
Then later click through to a different post
`https://ghost.site/other-post`

You'll essentially lose that first `rel` because the last one already will return some source and medium defaults. Is this intended behaviour?

Reply:
@SimonBackx Yea i see what you mean, we'll definitely update this to be more smart about how we calculate the referrer source/medium from the history, for example as you said we could maybe first check history for any entry that has `ref` and if not only then fallback to calculating them from URL. 
The current logic to calculate referrer sources just places the pieces in place for where things will happen, and we'll have more iterations on how the values are calculated from history once this is good :)

Reply:
Perfect! 👌 Just wanted to make sure 😊

Reply:
That's good, thanks 😄

Reply:
This should be much better now with updated logic to calculate referrer from history :)